src/IssueViz.Security.UnitTests/Taint/TaintIssuesSynchronizerTests.cs

```diff
@@ -61,38 +61,60 @@ public void MefCtor_CheckIsExported()
         }
 
         [TestMethod]
-        public void SynchronizeWithServer_FailureToCheckServerConnection_CriticalException_ExceptionThrown()
+        public async Task SynchronizeWithServer_NonCriticalException_ExceptionCaughtAndLogged()
         {
-            var sonarQubeServer = new Mock<ISonarQubeService>();
-            sonarQubeServer.Setup(x => x.GetServerInfo()).Throws(new StackOverflowException());
+            var logger = new TestLogger();
+
+            var sonarServer = CreateSonarService();
+            sonarServer.Setup(x => x.GetTaintVulnerabilitiesAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<CancellationToken>()))
+                .Throws(new Exception("this is a test"));
 
             var testSubject = CreateTestSubject(
                 bindingConfig: BindingConfig_Connected,
-                sonarService: sonarQubeServer.Object);
+                logger: logger,
+                sonarService: sonarServer.Object);
 
             Func<Task> act = testSubject.SynchronizeWithServer;
+            await act.Should().NotThrowAsync();
 
-            act.Should().Throw<StackOverflowException>();
+            logger.AssertPartialOutputStringExists("this is a test");
         }
 
         [TestMethod]
-        public void SynchronizeWithServer_FailureToCheckServerConnection_NonCriticalException_ExceptionIsCaught()
+        public async Task SynchronizeWithServer_NonCriticalException_UIContextAndStoreCleared()
```

Comment:
There are quite a few tests that check that `_UIContextAndStoreCleared` in various scenarios (e.g. non-critical exception, standalone mode, no issues returned).

Previously, some tests checked the UIContext, some checked StoreCleared, some checked both.
Now, all tests check both.